main/src/cgeo/geocaching/maps/CGeoMap.java

```diff
@@ -1316,28 +1300,22 @@ private void doDisplayRun() {
 
             // display caches
             final List<Geocache> cachesToDisplay = caches.getAsList();
-            final List<Waypoint> waypointsToDisplay = new ArrayList<>(waypoints);
```

Comment:
This was introduced in #1402 to avoid iterating the original waypoints collection asynchronously. Generally such copy constructors in cgeo should only be removed if one is completely sure there is no more concurrent access to the collection in question.